mongo/integration/client_side_encryption_prose_test.go

```diff
@@ -2059,7 +2059,142 @@ func TestClientSideEncryptionProse(t *testing.T) {
 			_, err = encClient.Database("db").Collection("coll").InsertOne(context.Background(), bson.D{{"unencrypted", "test"}})
 			assert.Nil(mt, err, "InsertOne error: %v", err)
 		})
+	autoKeyRunOpts := mtest.NewOptions().MinServerVersion("6.0").Topologies(mtest.ReplicaSet, mtest.Sharded, mtest.LoadBalanced, mtest.ShardedReplicaSet)
+	mt.RunOpts("21. automatic data encryption keys", autoKeyRunOpts, func(mt *mtest.T) {
+		setup := func() (*mongo.Client, *mongo.ClientEncryption, error) {
+			opts := options.Client().ApplyURI(mtest.ClusterURI())
+			client, err := mongo.Connect(context.Background(), opts)
+			if err != nil {
+				return nil, nil, err
+			}
+			client.Database("keyvault").Collection("datakeys").Drop(context.Background())
+			client.Database("db").Drop(context.Background())
+			ceo := options.ClientEncryption().
+				SetKmsProviders(fullKmsProvidersMap).
+				SetKeyVaultNamespace(kvNamespace)
+			clientEnc, err := mongo.NewClientEncryption(client, ceo)
+			if err != nil {
+				return nil, nil, err
+			}
+			return client, clientEnc, nil
+		}
+
+		mt.Run("case 1: simple creation and validation", func(mt *mtest.T) {
+			client, clientEnc, err := setup()
+			assert.Nil(mt, err, "setup error: %v", err)
+			defer func() {
+				client.Disconnect(context.Background())
+				clientEnc.Close(context.Background())
```

Comment:
```suggestion
				err := clientEnc.Close(context.Background())
				assert.Nil(mt, err, "error in Close")
```

`clientEnc.Close` already calls `Close` on the `keyVaultClient`.